packages/devtools/lib/src/utils.dart

```diff
@@ -321,3 +321,35 @@ class RateLimiter {
     _activeTimer?.cancel();
   }
 }
+
+// If the need arises, this enum can be expanded to include any of the remaining
+// time units supported by [Duration] - (milliseconds, seconds, etc.). If you
+// add a unit of time to this enum, modify the toString() method in [TimeRange]
+// to handle the new case.
+enum TimeUnit {
+  microseconds,
+  milliseconds,
+}
+
+class TimeRange {
+  TimeRange({this.unit = TimeUnit.microseconds, this.start, this.end});
+
+  final TimeUnit unit;
+
+  Duration start;
+  Duration end;
+
+  Duration get duration =>
+      Duration(microseconds: end.inMicroseconds - start.inMicroseconds);
+
+  @override
+  String toString() {
+    switch (unit) {
+      case TimeUnit.microseconds:
```

Comment:
this seems a bit surprising: is there a use case aside from tests?
The debug output does not clarify the unit which could lead to confusion if someone who didn't write this code is debugging the output.
I would suggest changing the toString to take an option TimeUnit named parameter
and having the output indicate the time unit if it is not the default unit.

Reply:
added optional param to toString().